### ISSUE TITLE
delete_ns blocks thread indefinitely

### ISSUE THREAD

**Issue by LambdaSpg** [Totodore/socketioxide#311]
**Describe the bug**
Calling delete_ns blocks the thread because the RWLock does not get released after going out of scope

**To Reproduce**
```rust
    let io_clone = Arc::new(io.clone());
    io.ns("/", move |socket_ref: SocketRef| {
        info!("Connected!");
        socket_ref.on_disconnect(move || {
            io_clone.delete_ns("/");
            info!("This won't run, because the thread is blocked by RWLock");
        })
    });
``` 

Connect a client on '/', disconnect it, and the info won't get called. After that, the server does not work anymore and won't accept new connections.

**Expected behavior**
The server should continue running like usual

**Versions (please complete the following information):**
 - Socketioxide version: 0.12.0
 - Http lib: 0.7.5
 - Socket.io client version ^4.7.5

**Comment by Totodore** [comment]
Hey, indeed, there is a deadlock when you call `delete_ns` from `on_disconnect` because the Namespace `RwLock` is currently being readen. I'll try to fix this ASAP and also add a test case.
